### PR TITLE
feat(macro): Adds custom derive attributes

### LINKED ISSUE #148: Adding custom derive attributes
Wit bindgen recently merged support for adding custom derive attributes: https://github.com/bytecodealliance/wit-bindgen/pull/678. Is that something that can be exposed in `cargo component`? I'm fine to do the work, just need a little bit of direction about where it would belong

### PR DESCRIPTION
**Pull request by thomastaylor312**
This adds custom derive attributes to the generate macro. It also fixes a bug where package version was not appended to the hash map of exports

Closes #148

### CODE REVIEW

**Comment by thomastaylor312** [comment]
@peterhuene This is only draft because right now it is depending on a git dependency of wit-bindgen until 0.13 can be released. That release also combined two of the rust crates (see https://github.com/bytecodealliance/wit-bindgen/pull/683), so I had to make those changes here as well. I did test this manually with some projects I was using, but let me know if I missed anything

File: crates/macro/src/lib.rs
Comment (peterhuene) [inline crates/macro/src/lib.rs:288]: Thanks for fixing this!

**Review by peterhuene — COMMENTED** [review]
The implementation looks great!

I think we should probably add a build test that applies a drive and tests it (like a simple serialization test).

**Comment by peterhuene** [comment]
Also, as this depends on an unreleased version of wit-bindgen, I'm going to hold off on merging until #147 is merged so that I can cut the first releases of the crates.

**Comment by thomastaylor312** [comment]
> I think we should probably add a build test that applies a drive and tests it (like a simple serialization test).

@peterhuene I didn't add one because there is already a test in wit-bindgen that tests for this. Did you still want to add one here as a smoke test (just making sure it can compile with the `additional_derives` value set)?

**Comment by thomastaylor312** [comment]
Ok, added a test anyway @peterhuene

**Comment by peterhuene** [comment]
> @peterhuene I didn't add one because there is already a test in wit-bindgen that tests for this. Did you still want to add one here as a smoke test (just making sure it can compile with the additional_derives value set)?

Yeah, it'd be good to have the test to cover that the argument to `cargo_component_bindings::generate!` actually works, in addition to the test coverage in wit-bindgen to ensure the generated bindings are correct.

> Ok, added a test anyway @peterhuene

Thanks!

**Review by peterhuene — APPROVED** [review]

**Comment by thomastaylor312** [comment]
I'll leave this in draft to avoid accidental merge until either

a) You've cut a version of this to release the crate
b) wit-bindgen releases a new tagged version

**Comment by peterhuene** [comment]
We're now good to rebase this PR and change it from a draft; we don't need to wait for a wit-bindgen release.

**Comment by thomastaylor312** [comment]
Ok, I'll rebase!

**Comment by thomastaylor312** [comment]
@peterhuene This should be good to go

**Comment by peterhuene** [comment]
@thomastaylor312 thanks!

**Pull request merged**
